optional_plugins/mail/avocado_mail_result/mail_result.py

```diff
@@ -0,0 +1,257 @@
+    def _format_test_details(test, advanced=False):
+        if advanced:
+            details = [
+                f"<strong>Name:</strong> {test.get('name', '')}<br>",
+                f"<strong>Status:</strong> {test.get('status', '')}<br>",
+                f"<strong>Fail Reason:</strong> {test.get('fail_reason', '')}<br>",
+                f"<strong>Actual Time Start:</strong> {test.get('actual_time_start', '')}<br>",
+                f"<strong>Actual Time End:</strong> {test.get('actual_time_end', '')}<br>",
+                f"<strong>ID:</strong> {test.get('id', '')}<br>",
+                f"<strong>Log Directory:</strong> {test.get('logdir', '')}<br>",
+                f"<strong>Log File:</strong> {test.get('logfile', '')}<br>",
+                f"<strong>Time Elapsed:</strong> {test.get('time_elapsed', '')}<br>",
+                f"<strong>Time Start:</strong> {test.get('time_start', '')}<br>",
+                f"<strong>Time End:</strong> {test.get('time_end', '')}<br>",
+                f"<strong>Tags:</strong> {test.get('tags', '')}<br>",
+                f"<strong>Whiteboard:</strong> {test.get('whiteboard', '')}<br>",
+            ]
+        else:
+            details = [
+                f"<strong>Name:</strong> {test.get('name', '')}<br>",
+                f"<strong>Fail Reason:</strong> {test.get('fail_reason', '')}<br>",
+            ]
+        return "".join(details)
+
+    @staticmethod
+    def _generate_test_summary(data, detail_level):
+        test_summary = []
+
+        def format_test_details(test):
+            return Mail._format_test_details(test, advanced=detail_level)
+
+        for test in data.get("tests", []):
+            if test.get("status") == "FAIL":
+                test_summary.append(format_test_details(test))
+
+        return "\n\n".join(test_summary)
+
+    @staticmethod
+    def _get_test_summary(job):
+        results_path = os.path.join(job.logdir, "results.json")
+        data = Mail._read_results_file(results_path)
+        if not data:
+            return ""
+
+        detail_level = job.config.get("plugins.mail.detail_level")
+        return Mail._generate_test_summary(data, detail_level)
+
+    def send_start_email(self, job):
+        phase = "Start"
+        server, port, sender, password = Mail._get_smtp_config(job)
+        smtp = Mail._create_smtp_connection(server, port)
+        if smtp:
+            try:
+                smtp.login(sender, password)
+            except Exception as e:  # pylint: disable=W0703
+                LOG_UI.error(f"SMTP login failed: {e}")
+                return
+            start_time = time.strftime("%Y-%m-%d %H:%M:%S", time.localtime())
+            time_content = f"{start_time}"
+
+            msg = Mail._build_message(job, time_content, phase)
+            Mail._send_email(
+                smtp, sender, job.config.get("plugins.mail.recipient"), msg
+            )
+            smtp.quit()
+
+    def pre(self, job):
+        self.send_start_email(job)
```

Comment:
I would move the `send_start_email` method code in here, same as it is in `post` method for finish message.